yt/frontends/boxlib/fields.py

```diff
@@ -493,8 +493,10 @@ def setup_fluid_fields(self):
 
                 # Most of the time our species will be of the form
                 # element name + atomic weight (e.g. C12), but
-                # sometimes we make up descriptive names (e.g. ash)
-                if any(char.isdigit() for char in field):
+                # sometimes we make up descriptive names (e.g. ash).
+                # In niche cases, we might have a molecule with a number
+                # in the middle (e.g. H2O). We ignore those.
+                if any(char.isdigit() for char in field) and field[-2].isdigit():
```

Comment:
I don't think I understand what this branch is supposed to capture. Both examples in the comment (`C12`, `H20`) check `field[-2].isdigit()`, so they both go through, but what I understand from the comment is that `C12` should go, and `H2O` shouldn't. What am I missing ?

Reply:
Actually the strings in the plotfiles are X(C12), X(H2O).  Previously the last character before the ")" parenthesis would be a number, but not in the H2O case, which will raise an error in `weight = int(weight)` a few lines down.  Note that this block of code is for the moment superfluous, element and weight are not used.

Reply:
The comment should reflect that then. Ideally the code would be expressive enough that a comment wouldn't even be needed.
Are you saying the whole block is superfluous ? If so, I think it should be removed, or at least not touched in this PR.

Reply:
I think it was put in there so eventually we could use the element and weight variables (@zingale ?)
https://github.com/yt-project/yt/blob/d7209f9aab2587bca4264f42c2106d4bf6b5cf88/yt/frontends/boxlib/fields.py#L509-L510

Reply:
I only made this change so that the code wouldn't break when there is a species like H2O.

Reply:
wouldn't the ideal fix be to extend the regexp so that it doesn't break for these cases ?

Reply:
Yes, I mentioned this when I submitted the PR.  I tried for a while but it was too difficult for me.  Ideally you would also make the numbers subscripts instead of superscript for the tex label for molecules.  Again, having molecules in the first place is a very niche use of castro/maestro...

Reply:
Ok, thanks for clarifying. I suggest we slightly rephrase the logging error message to make it clear that it's a missing feature, not an actual bug, so that readers might choose to give it a try too :)

Reply:
I wrote a brief something, but please feel free to edit